dynamodb/dynamodb_test.go

```diff
@@ -109,48 +157,75 @@ func Test_DynamoDB_Reset(t *testing.T) {
 }
 
 func Test_DynamoDB_Close(t *testing.T) {
+	testStore, err := newTestStore(t)
+	require.NoError(t, err)
+
 	require.Nil(t, testStore.Close())
 }
 
 func Test_DynamoDB_Conn(t *testing.T) {
+	testStore, err := newTestStore(t)
+	require.NoError(t, err)
+	defer testStore.Close()
+
 	require.True(t, testStore.Conn() != nil)
 }
 
 func Benchmark_DynamoDB_Set(b *testing.B) {
 	b.ReportAllocs()
 	b.ResetTimer()
 
-	var err error
+	testStore, err := newTestStore(b)
+	require.NoError(b, err)
+	defer testStore.Close()
+
+	errs := make([]error, b.N)
 	for i := 0; i < b.N; i++ {
-		err = testStore.Set("john", []byte("doe"), 0)
+		errs[i] = testStore.Set("john", []byte("doe"), 0)
 	}
 
-	require.NoError(b, err)
+	for _, err := range errs {
+		require.NoError(b, err)
+	}
 }
 
 func Benchmark_DynamoDB_Get(b *testing.B) {
-	err := testStore.Set("john", []byte("doe"), 0)
+	testStore, err := newTestStore(b)
+	require.NoError(b, err)
+	defer testStore.Close()
+
+	err = testStore.Set("john", []byte("doe"), 0)
 	require.NoError(b, err)
 
 	b.ReportAllocs()
 	b.ResetTimer()
 
+	errs := make([]error, b.N)
 	for i := 0; i < b.N; i++ {
 		_, err = testStore.Get("john")
+		errs[i] = err
```

Comment:
We don't need this here, since these are benchmarks not tests.

Reply:
question: I'm not that familiar with how benchmarks deal with errors. Would the for loop be hiding errors that happen in between, just capturing if the last error was a real error?

Reply:
You have 3 options:
- ignore the errors
- capture the last one and check outside
- add a separate benchmark that checks on each loop (this will be very slow but validates that each loop is working as intended.

I have implemented the last case in the templates repo, it was used to find a data race issue. See here https://github.com/gofiber/template/blob/master/html/html_test.go

For storage the DB itself handles concurrency, we should be fine with just ignoring the error.

If we wanted to in the future all the benchmarks can be updated to look like the templates repo

Reply:
Cool, thanks for the explanation, will update this PR now